### PR TITLE
Add script to mask off frozen regions of hydro domain

### PR DESCRIPTION
**Pull request by alexolinhager**
Creates a new script, floodFillThawedIce.py, that deactivates frozen regions of the hydro domain by using a flood fill of thawed basal ice originating at the grounding line. Thickness of frozen ice is set to zero and a no-flow waterFluxMask is defined surrounding the frozen area.

### CODE REVIEW

**Comment by xylar** [comment]
@alexolinhager, if you want this to be available in the `mpas_tools` conda package, it will also need to be added here:
https://github.com/MPAS-Dev/MPAS-Tools/blob/master/conda_package/setup.py#L70
and ideally you would add some sort of test to make sure it exists somewhere around here:
https://github.com/MPAS-Dev/MPAS-Tools/blob/master/conda_package/recipe/meta.yaml#L121

A final note that we typically use script names that are lowercase with underscores but I'll leave it up to @matthewhoffman how strict things are with the land ice scripts.

**Comment by alexolinhager** [comment]
Thanks @xylar, right now we just want this to be a script we can refer to and will use occasionally, but that doesn't need to be part of the conda package. I'll change the name though to make sure it's consistent with the rest of MPAS-Tools

**Comment by xylar** [comment]
Okay, sounds good!

File: landice/mesh_tools_li/floodFillThawedIce.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env python
+
```
Comment (matthewhoffman) [inline landice/mesh_tools_li/floodFillThawedIce.py:2]: Add a description of what the script does inside of a block of 
```
'''
...description...
'''
```

**Review by matthewhoffman — CHANGES_REQUESTED** [review]
@alexolinhager , thanks for opening a PR for this script.  It will be good to have this saved for future use.  I have requested some changes.  A few will change the result, so take a careful look at things after making the adjustments and make sure those suggestions make sense.

File: landice/mesh_tools_li/floodFillThawedIce.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+seedMask = np.zeros((len(nEdgesOnCell),1), 'float64')
+
+print("**Defining Seed and Grow Masks ...")
+ind = np.where(groundedMask==1)[0]
+for i in ind:
+    #identify grounded cells just inland of grounding line
+    for ii in range(nEdgesOnCell[i]):
+      # if ((groundedMask[cellsOnCell[i,ii]-1] == 0):
+       if ((floatingMask[cellsOnCell[i,ii]-1] == 1) | (oceanMask[cellsOnCell[i,ii]-1] == 1)):
```
[the remark this answers is not in the document]
Comment (alexolinhager) [reply landice/mesh_tools_li/floodFillThawedIce.py:47]: This part is fairly quick, especially compared to `mpas_flood_fill`, so may just keep this as is

File: landice/mesh_tools_li/floodFillThawedIce.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+          seedMask[i] = 1
+
+#identify grounded, thawed ice
+basalSlidingSpeed = np.sqrt(uReconstructX**2 + uReconstructY**2) * 3.15e7 #convert to m/yr
+
+growMask = (abs(groundedBasalMassBal) > 0.0)*(basalSlidingSpeed > UbThresh)
```
[the remark this answers is not in the document]
Comment (alexolinhager) [reply landice/mesh_tools_li/floodFillThawedIce.py:53]: Removed the abs() from the equations and changed this to np.logical_or()

File: landice/mesh_tools_li/floodFillThawedIce.py [hunk 4]
```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env python
+'''
+Uses output from MALI's thermal solver to determine where regions of the bed are frozen or thawed and masks of frozen cells from the hydro domain.
```
Comment (matthewhoffman) [inline landice/mesh_tools_li/floodFillThawedIce.py:3]:
```suggestion
Uses output from MALI's thermal solver to determine where regions of the bed are frozen or thawed and masks off frozen cells from the hydro domain.
```

**Review by matthewhoffman — CHANGES_REQUESTED** [review]
@alexolinhager , thanks for the changes.  I re-reviewed and just noticed one minor typo I made a suggestion for.  Once you fix that and rename the script I can go ahead and merge this.  

One other thing - can you comment on what impact the changes to the logic had? (changing growMask to use an `or` of the two conditions; considering thawed regions only where BMB is negative rather than nonzero)

**Comment by alexolinhager** [comment]
@matthewhoffman I made a few more changes to the script, mostly just minor debugging to allow the script to work with different input files that may have various variables in it that we want to get rid of. I did change back the calculation of `basalMeltInput` to `abs(groundedBasalMassBal)` though. For some reason I don't understand the previous implementation was creating all zeros throughout the domain. It seems like groundedBasalMassBal should almost always be negative, so it doesn't seem like this would have too much of an detrimental effect.

The change from `and` to `or` in the growMask definition was certainly necessary. Judging by the figure below (thawed areas are outlined in black), using an `and` requirement would have limited the active hydro domain to the small regions where both thawed ice and basalSlidingSpeed > 25 m/yr overlap, which would drastically cut down on the active hydro domain.
Screenshot 2024-03-22 at 2 47 30 PM

Based on the plots below though, I am confident the current script is doing what we want it to do. Non-zero thickness only exists in areas where basalMeltInput > 0, basalSlidingSpeed > 25 m/yr, and that are connected to the grounding line. Using the additional basalSlidingSpeed criteria allows us to capture the Whillans Ice Stream while having a minimal effect on the rest of the domain
Screenshot 2024-04-09 at 5 46 33 PM
Screenshot 2024-04-09 at 5 46 48 PM
Screenshot 2024-04-09 at 5 47 11 PM

**Review by matthewhoffman — APPROVED** [review]
@alexolinhager, looks good, thanks for making the changes

**Pull request merged**